url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/io/Generic_StaticIO.java

```diff
@@ -0,0 +1,2156 @@
+    }
+
+    /**
+     * @param f
+     * @return <code>new ObjectInputStream(getBufferedInputStream(f)</code>
+     */
+    public static ObjectInputStream getObjectInputStream(File f) {
+        ObjectInputStream result = null;
+        BufferedInputStream bis;
+        bis = getBufferedInputStream(f);
+        try {
+            result = new ObjectInputStream(bis);
+        } catch (IOException ex) {
+            Logger.getLogger(Generic_StaticIO.class.getName()).log(Level.SEVERE, null, ex);
+        }
+        return result;
+    }
+
+    /**
+     * @param f
+     * @return <code>new ObjectOutputStream(getBufferedOutputStream(f)</code>
+     */
+    public static ObjectOutputStream getObjectOutputStream(File f) {
+        ObjectOutputStream result = null;
+        BufferedOutputStream bos;
+        bos = getBufferedOutputStream(f);
+        try {
+            result = new ObjectOutputStream(bos);
+        } catch (IOException ex) {
+            Logger.getLogger(Generic_StaticIO.class.getName()).log(Level.SEVERE, null, ex);
+        }
+        return result;
+    }
+
+    private static void copyDirectory(
+            File directoryToCopy,
+            File directoryToCopyInto) {
+        try {
+            if (!directoryToCopyInto.mkdir()) {
+                directoryToCopyInto.mkdirs();
+            }
+            if (!directoryToCopy.isDirectory()) {
+                throw new IOException(
+                        "Expecting File " + directoryToCopy
+                        + " To be a directory in "
+                        + Generic_StaticIO.class.getName()
+                        + ".copyDirectory(File,File)");
+            }
+            if (!directoryToCopyInto.isDirectory()) {
+                throw new IOException(
+                        "Expecting File " + directoryToCopyInto
+                        + " To be a directory in "
+                        + Generic_StaticIO.class.getName()
+                        + ".copy(File,File)");
+            }
+            File copiedDirectory_File = new File(
+                    directoryToCopyInto,
+                    directoryToCopy.getName());
+            copiedDirectory_File.mkdir();
+            File[] a_DirectoryToCopy_File_Files
+                    = directoryToCopy.listFiles();
+            for (File f : a_DirectoryToCopy_File_Files) {
+                if (f.isFile()) {
+                    copyFile(f, copiedDirectory_File);
+                } else {
+                    copyDirectory(f, copiedDirectory_File);
+                }
+            }
+        } catch (IOException e) {
+            System.err.print(e.getMessage());
+            e.printStackTrace(System.err);
+            System.exit(Generic_ErrorAndExceptionHandler.IOException);
+        }
+    }
+
+    public static void copy(
+            File fileOrDirectoryToCopy,
+            File directoryToCopyInto) {
+        try {
+            if (!directoryToCopyInto.mkdir()) {
+                directoryToCopyInto.mkdirs();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Confirm that `mkdirs` was successful. Handle any failures that happened. If you fail silently, then you might encounter errors that are harder to debug. See the following diffs on GitHub that make this code change: [diff1](https://github.com/qunarcorp/qmq/commit/12f1d9d6a57a02bc2415cb2af9f4388a77b12c02#diff-6b67be6e17f24fdeddb7c725ac8ec5600d6e63be0bdca0963a83d530e6d55e35L36), [diff2](https://github.com/apache/jmeter/commit/1ec61206779b3f84df18516682b4791bf015178f#diff-cd74304763c097fbc085ac2bc952b46ab3843ac825cacacad89e8064296e955aL330)

Learn more: [File.mkdirs in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#mkdirs--)